x-pack/plugins/security_solution/public/common/hooks/use_upgrade_secuirty_packages.test.tsx

```diff
@@ -65,4 +69,66 @@ describe.skip('When using the `useUpgradeSecurityPackages()` hook', () => {
       })
     );
   });
+
+  it('should send upgrade request WITHOUT prerelease:true if branch is not `main` and build is not `-SNAPSHOT`', async () => {
+    mockGetKibanaVersion.mockReturnValue('8.0.0');
+    mockGetKibanaBranch.mockReturnValue('release');
+
+    renderHook();
+
+    await renderResult.waitFor(
+      () => (kibana.services.http.post as jest.Mock).mock.calls.length > 0
+    );
+
+    expect(kibana.services.http.post).toHaveBeenCalledWith(
+      `${epmRouteService.getBulkInstallPath()}`,
+      expect.objectContaining({
+        body: '{"packages":["endpoint","security_detection_engine"]}',
+        query: expect.objectContaining({ prerelease: false }),
+      })
+    );
+  });
+
+  it('should send upgrade request with prerelease:true if branch is `main` or build is `-SNAPSHOT`', async () => {
+    mockGetKibanaVersion.mockReturnValue('8.0.0-SNAPSHOT');
+    mockGetKibanaBranch.mockReturnValue('main');
+
+    renderHook();
+
+    await renderResult.waitFor(
+      () => (kibana.services.http.post as jest.Mock).mock.calls.length > 0
+    );
+
+    expect(kibana.services.http.post).toHaveBeenCalledWith(
+      `${epmRouteService.getBulkInstallPath()}`,
+      expect.objectContaining({
+        body: '{"packages":["endpoint","security_detection_engine"]}',
+        query: expect.objectContaining({ prerelease: true }),
+      })
+    );
+  });
```

Comment:
Nit: to make sure the OR logic works, we should probably have two unit tests:

1: `8.0.0-SNAPSHOT` + `release`
2: `8.0.0` + `main`